fluxc/src/main/java/org/wordpress/android/fluxc/store/CommentsStore.kt

```
@@ -677,20 +678,25 @@ class CommentsStore @Inject constructor(
                     "Comments Unification project proceeds",
             replaceWith = ReplaceWith("use createNewComment suspend fun directly")
     )
+    @Suppress("UseCheckOrError")
     private suspend fun onCreateNewComment(payload: RemoteCreateCommentPayload): OnCommentChanged {
-        val response = if (payload.reply == null) {
+        val response = if (payload.post != null && payload.reply == null) {
             // Create a new comment on a specific Post
             createNewComment(
                     payload.site,
                     commentsMapper.commentLegacyModelToEntity(payload.comment)
             )
-        } else {
+        } else if (payload.reply != null && payload.post == null) {
             // Create a new reply to a specific Comment
             createNewReply(
                     payload.site,
                     commentsMapper.commentLegacyModelToEntity(payload.comment),
                     commentsMapper.commentLegacyModelToEntity(payload.reply)
             )
+        } else {
+            throw IllegalStateException(
+                "Either post or reply must be not null and both can't be not null at the same time!"
+            )
```

This will likely crash client apps in production if this scenario happens - clients don't have possibility to handle this exception. Can we change it to a log and failed `OnCommentChange` event?

---

👋 @wzieba and thanks for this comment, this is indeed an interesting one, let me explain:

1. First, you can take a look at this commit's description: 53e14a0bde415d92fede132f261b3161f843e846
2. From there, you will notice that I did that because the two `RemoteCreateCommentPayload` constructors are also defined as such and thus the callers should also respect their contract in order to avoid any NPE or an illegal app state. I also chose to keep the `payload.reply == null` condition and add the extra `payload.post == null` condition because the `payload.reply == null` condition was already present.
3. So, all in all, we should never reach this `else` case and thus if we do it is end-up being an illegal state of the app. Instead of trying to hide this state and somehow mute it I think it is better to throw an exception, that is, if and whenever someone changes how the `RemoteCreateCommentPayload` is constructed, because if they don't, as it is now, it will never happen anyway.

Does that makes sense? 🙏

---

> the two RemoteCreateCommentPayload constructors are also defined as such and thus the callers should also respect their contract in order to avoid any NPE or an illegal app state

Thanks. When writing this comment, I was under the impression that `RemoteCreateCommentPayload` is a mapped response from the API. But then I looked at `WordPress-Android` codebase and understood, that it's a request created via constructor. Let's keep the exception, then!